rotkehlchen/inquirer.py

```diff
@@ -500,6 +536,154 @@ def find_uniswap_v2_lp_price(
+        contract = EthereumContract(
+            address=pool.pool_address,
+            abi=CURVE_POOL_ABI,
+            deployed_block=0,
+        )
+        calls = [(pool.pool_address, contract.encode(method_name='get_virtual_price'))]
+        calls += [
+            (pool.pool_address, contract.encode(method_name='balances', arguments=[i]))
+            for i in range(len(pool.assets))
+        ]
+        output = multicall_2(
+            ethereum=self._ethereum,
+            require_success=False,
+            calls=calls,
+        )
+
+        # Check that the output has the correct structure
+        if not all([len(call_result) == 2 for call_result in output]):
+            log.debug(
+                f'Failed to query contract methods while finding curve pool price. '
+                f'Not every outcome has length 2. {output}',
+            )
+            return None
+        # Check that all the request were successful
+        if not all([contract_output[0] for contract_output in output]):
+            log.debug(f'Failed to query contract methods while finding curve price. {output}')
+            return None
+        # Deserialice information obtained in the multicall execution
+        data = []
+        virtual_price_decoded = contract.decode(output[0][1], 'get_virtual_price')
+        # Verify that the output from the contrat call is correct
+        if not _check_curve_contract_call(virtual_price_decoded):
+            log.debug(f'Failed to decode get_virtual_price while finding curve price. {output}')
+            return None
+        data.append(FVal(virtual_price_decoded[0]))
+        for i in range(len(pool.assets)):
+            amount_decoded = contract.decode(output[i + 1][1], 'balances', arguments=[i])
+            # Verify that the contract decoded the output correctly
+            if not _check_curve_contract_call(amount_decoded):
+                log.debug(f'Failed to decode balances {i} while finding curve price. {output}')
+                return None
+            amount = amount_decoded[0]
+            normalized_amount = token_normalized_value_decimals(amount, tokens[i].decimals)
+            data.append(normalized_amount)
+
+        # Prices and data should verifty this relation for the following operations
+        if len(prices) != len(data) - 1:
+            log.debug(
+                f'Length of prices {len(prices)} does not match len of data {len(data)} '
+                f'while querying curve pool price.',
+            )
+            return None
+        # Total number of assets price in the pool
+        total_assets_price = sum(map(operator.mul, data[1:], prices))
+        if total_assets_price == 0:
+            log.error(
+                f'Curve pool price returned unexpected data {data} that lead to a zero price.',
+            )
+            return None
+
+        # Calc weight of each asset as the proportion of tokens value
```

Comment:
absolutely no reason to abbreviate the verb here

```suggestion
        # Calculate weight of each asset as the proportion of tokens value
```